lib/dolly/collection.rb

```diff
@@ -1,7 +1,7 @@
 require 'delegate'
 
 module Dolly
-  class Collection < DelegateClass(Array)
+  class Collection < SimpleDelegator
```

Comment:
there's no issue with previous versions of ruby with this change?

Reply:
It behaves the same and SimpleDelegator is available since ruby 2.4 if I'm not wrong.

Reply:
If this changes rules out some versions of Ruby, then that should be reflected in the version... This should probably be done on a new branch or something as it would be a breaking change for applications using older rubies